lib/Conversion/TorchToLinalg/Random.cpp

```diff
@@ -154,6 +160,104 @@ class ConvertAtenUniformOp : public OpConversionPattern<AtenUniformOp> {
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+    Location loc = op.getLoc();
+    RankedTensorType resultType = getTypeConverter()
+                                      ->convertType(op->getResult(0).getType())
+                                      .cast<RankedTensorType>();
+    Type resultElemTy = resultType.getElementType();
+
+    // The pin_memory should be either `False` or `none`.
+    bool pinMemory;
+    if (!op.pin_memory().getType().template isa<Torch::NoneType>() &&
+        (!matchPattern(op.pin_memory(), m_TorchConstantBool(&pinMemory)) ||
+         pinMemory)) {
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: pin_memory must be either None or false");
+    }
+
+    SmallVector<int64_t> outSize;
+    if (!matchPattern(op.size(), m_TorchListOfConstantInts(outSize)))
+      return rewriter.notifyMatchFailure(op,
+                                         "only support constant int size list");
+
+    Value initTensor =
+        rewriter.create<tensor::EmptyOp>(loc, outSize, resultElemTy);
+
+    Value multiplier = rewriter.create<arith::ConstantOp>(
+        loc, rewriter.getI64IntegerAttr(6364136223846793005));
+    Value incrementStep = rewriter.create<arith::ConstantOp>(
+        loc, rewriter.getI64IntegerAttr(1442695040888963407));
+
+    // Get initial seed, min and max used by `linalg.generic` compute payload.
+    Value initialSeed = rewriter.create<TorchConversion::GetNextSeedOp>(loc);
+    Value min = rewriter.create<arith::ConstantOp>(
+        loc, rewriter.getFloatAttr(resultElemTy, 0.0));
+    Value max = rewriter.create<arith::ConstantOp>(
+        loc, rewriter.getFloatAttr(resultElemTy, 1.0));
+
+    Value cstMinusTwo = rewriter.create<arith::ConstantOp>(
+        loc, rewriter.getFloatAttr(resultElemTy, -2.0));
+    Value cstTwoPie = rewriter.create<arith::ConstantOp>(
+        loc, rewriter.getFloatAttr(resultElemTy, 2.0 * 3.14159));
+
+    // Construct the `linalg.generic` op.
+    auto resultRank = resultType.getRank();
+    SmallVector<AffineMap, 1> indexingMaps(
+        1, rewriter.getMultiDimIdentityMap(resultRank));
+    SmallVector<utils::IteratorType> iteratorTypes(
+        resultRank, utils::IteratorType::parallel);
+    Value randnRes =
+        rewriter
+            .create<linalg::GenericOp>(
+                loc, initTensor.getType(), /*inputs=*/ValueRange{},
+                /*outputs=*/initTensor, indexingMaps, iteratorTypes,
+                [&](OpBuilder &b, Location loc, ValueRange args) {
+                  Value temp = initialSeed;
+                  for (int i = 0; i < resultRank; i++) {
+                    Value index = b.create<linalg::IndexOp>(loc, i);
+                    temp = getNextTemp(b, loc, index, temp, multiplier,
+                                       incrementStep);
+                  }
```

Comment:
I would include this `for` loop in `getNextTemp`. It makes sense for the entire LCG algorithm to be in a single function.